app/models/steps/prefill.rb

```diff
@@ -0,0 +1,37 @@
+module Steps
+  class Prefill < Steps::Base
+    include CommodityHelper
+
+    def next_step_path
+      return duty_path if user_session.ni_to_gb_route? || user_session.eu_to_ni_route?
+
+      return next_step_for_gb_to_ni if user_session.gb_to_ni_route?
+
+      return next_step_for_row_to_gb if user_session.row_to_gb_route?
+
+      next_step_for_row_to_ni
+    end
```

Comment:
I am not a big fan of cascade IFs, it makes this hard to follow,
and actually, this is hard to follow 😄 for me
I've no idea how to make it clear at the moment, I'll thy to think something 🤔

Reply:
I don't think there is a way around it, sadly. The flow of the routing for the duty calculator is just complex :(